src/fibsem_tools/io/fibsem_h5.py

```diff
@@ -0,0 +1,314 @@
+"""
+Utilities for loading and saving FIBSEM Data as HDF5 fliles
+
+High-level Interface
+* create_default_h5_files
+* create_zstd_h5_files
+* create_gzip_h5_files
+"""
+
+import h5py
+import numpy as np
+from typing import Iterable, Union, Sequence, Tuple
+import re
+import os.path
+import glob
+import hdf5plugin
+from .fibsem import (
+    FibsemDataset,
+    FIBSEMData,
+    FIBSEMHeader,
+    OFFSET,
+    MAGIC_NUMBER,
+    read_fibsem,
+)
+
+chunk_type = Union[Tuple[int, int], Tuple[int, int, int]]
+
+
+def create_fibsem_h5_dataset(
+    parent: Union[h5py.File, h5py.Group],
+    name: str,
+    data: FIBSEMData,
+    chunks: chunk_type = (128, 128),
+    **kwargs
+):
+    """
+    Create a HDF5 dataset from an existing FIBSEMData object.
+
+    :param parent: `h5py.File` or `h5py.Group` to place the dataset in
+    :param name: Name of the dataset
+    :param data: `FIBSEMData` instance
+    :param chunks: tuple describing the chunk size (Y, X)
+    :type chunks: tuple, optional
+    :param **kwargs: Keyword arguments are forwarded to `parent.create_dataset`
+    """
+
+    # If chunks has length 2, add a 1 for the ChanNum dimension
```

Comment:
why are we not allowing chunking along the channel axis?

Reply:
@StephanPreibisch mentioned earlier that the two channels were likely not related in intensity, so we should keep the channels as separate chunks. We should test this empirically.

Reply:
We may need to modify the `chunks` parameter here to also allow for the simple contiguous layout (non-chunked) case since this is meant as a low-level function. At the moment this would be useful mainly for benchmarking purposes.

The h5py docs suggest that setting this to `False` may do the trick though.

Reply:
why not just match the h5py API exactly and have the user put `chunks` into `**kwargs`

Reply:
I'm still working around the ambiguity of whether we are rolling the axes or not. I guess we should just commit to not rolling the axes.

Reply:
`chunks = None` disables chunking via `h5py`